volume/profiles/profiles.go

```diff
@@ -135,16 +111,16 @@ func GetVolProProfileByName(name string, pvc *v1.Volume) (VolumeProvisionerProfi
 // NOTE:
 //    This is a concrete implementation of
 // volumeprovisioner.VolumeProvisionerProfile
-type pvcVolProProfile struct {
-	pvc     *v1.Volume
+type defVolProProfile struct {
+	vol     *v1.Volume
 	vsmName string
 }
 
-// newPvcVolProProfile provides a new instance of VolumeProvisionerProfile that is
-// based on pvc (i.e. persistent volume claim).
-func newPvcVolProProfile(pvc *v1.Volume) (VolumeProvisionerProfile, error) {
-	return &pvcVolProProfile{
-		pvc: pvc,
+// newDefVolProProfile provides a new instance of VolumeProvisionerProfile that is
+// based on volume
+func newDefVolProProfile(vol *v1.Volume) (VolumeProvisionerProfile, error) {
```

Comment:
When will this method throw an `error` ? If its not throwing any error then why it is returning error?